features/0685-pickup-v2/README.md

```diff
@@ -0,0 +1,184 @@
+
+Message delivery is confirmed with a `message_received` acknowledgement, at which point the message can be removed from the mediator's queue.
+
+When live delivery is enabled, messages that arrive when an existing connection exists are delivered over the connection immediately.
+
+
+
+## Reference
+
+Each message sent should use the ~transport decorator as follows. This has been ommitted from the examples for brevity.
+
+```json=
+"~transport": {
+    "return_route": "all"
+}
+```
+
+
+### StatusRequest
+
+Sent by the _recipient_ to the _mediator_ to request a `status` message.
+
+```json=
+{
+    "@id": "123456781",
+    "@type": "https://didcomm.org/messagepickup/2.0/status-request",
+    "recipient_key": "<key for messages>"
+}
+```
+
+- `recipient_key` is optional. When specified only return status related to that recipient key. This allows the _Recipient_ to discover if any messages are in the queue that were sent to a specific key.
+
+### Status
+
+Status details about pending messages
+
+```json=
+{
+    "@id": "123456781",
+    "@type": "https://didcomm.org/messagepickup/2.0/status",
+    "recipient_key": "<key for messages>",
+    "message_count": 7,
+    "duration_waited": 3600,
+    "newest_time": "2019-05-01 12:00:00Z",
+    "oldest_time": "2019-05-01 12:00:01Z",
+    "total_size": 8096,
+    "live_delivery": false
+}
+```
+
+`message_count` is the only required attribute. The others may be present if offered by the _message_holder_.
+
+`duration_waited` is in seconds, and is the longest  delay of any message in the queue.
+
+`total_size` is in bytes.
+
+If a `recipient_key` was specified in the status-request message, the matching value MUST be specified in the `recipient_key` attribute of the status message.
+
+`live_delivery` state is also indicated in the status message. 
+
+### Delivery Request
+
+A request from the _Recipient_ to the _Mediator_ to have waiting messages delivered. 
+Examples:
+
+```json=
+{
+    "@id": "123456781",
+    "@type": "https://didcomm.org/messagepickup/2.0/delivery-request",
+    "limit": 10,
+    "recipient_key": "<key for messages>"
+}
+```
+
+```json=
+{
+    "@type": "https://didcomm.org/messagepickup/2.0/delivery-request",
+    "limit": 1
+}
+```
+
+
+After receipt of this message, the mediator should deliver up to the limit indicated. 
+
+`recipient_key` is optional. When specified only return messages sent to that recipient key.
+
+If no messages are available to be sent, a status message MUST be sent immediately.
+
+Delivered messages MUST not be deleted until delivery is acknowledged by a `messages-received` message.
+
+
+### Message Received
+After receiving messages, the Recipient sends an ack message like this:
+```json=
+{
+    "@type": "https://didcomm.org/messagepickup/2.0/messages-received",
+    "message_tag_list": ["123","456"]
+}
+```
+
+`message_tag_list` identifiers are tags of each message received, extracted from the encrypted form of each message.
```

Comment:
I believe you mentioned using a hash here. I would like some comment about what a "tag" means, and how one is found. If it's not a hash, what is it, and how does the mediator know it?

Reply:
the tag is a portion of a hash, but that detail isn't relevant to the implementation of the protocol.